cmake 4.0.x errors out on cargo install

**Describe the bug**

Attempt to package install oculante eventually errors out during cargo compilation of dependencies with cmake v4.0.x

Build does complete successfully with cmake v3.30.x.

**To Reproduce**

cargo install oculante

**Expected behavior**

Clean install.

**Screenshots / sample files**

```
...
   Compiling notan_core v0.12.1
error: failed to run custom build command for `glsl-to-spirv v0.1.7`

Caused by:
  process didn't exit successfully: `/home/shum/tmp/cargo-install4gkASx/release/build/glsl-to-spirv-b85ea852dad5a143/bui
ld-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-changed=build/glslangValidator.exe
  CMAKE_TOOLCHAIN_FILE_x86_64-unknown-linux-gnu = None
  CMAKE_TOOLCHAIN_FILE_x86_64_unknown_linux_gnu = None
  HOST_CMAKE_TOOLCHAIN_FILE = None
  CMAKE_TOOLCHAIN_FILE = None
  CMAKE_GENERATOR_x86_64-unknown-linux-gnu = None
  CMAKE_GENERATOR_x86_64_unknown_linux_gnu = None
  HOST_CMAKE_GENERATOR = None
  CMAKE_GENERATOR = None
  CMAKE_PREFIX_PATH_x86_64-unknown-linux-gnu = None
  CMAKE_PREFIX_PATH_x86_64_unknown_linux_gnu = None
  HOST_CMAKE_PREFIX_PATH = None
  CMAKE_PREFIX_PATH = None
  CMAKE_x86_64-unknown-linux-gnu = None
  CMAKE_x86_64_unknown_linux_gnu = None
  HOST_CMAKE = None
  CMAKE = None
  running: cd "/home/shum/tmp/cargo-install4gkASx/release/build/glsl-to-spirv-9e2f8516f02731a8/out/build" && CMAKE_PREFI
X_PATH="" LC_ALL="C" "cmake" "/home/shum/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/glsl-to-spirv-0.1.7/glslan
g" "-DCMAKE_INSTALL_PREFIX=/home/shum/tmp/cargo-install4gkASx/release/build/glsl-to-spirv-9e2f8516f02731a8/out" "-DCMAKE
_C_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_C_COMPILER=/usr/bin/cc" "-DCMAKE_CXX_FLAGS= -ffunctio
n-sections -fdata-sections -fPIC -m64" "-DCMAKE_CXX_COMPILER=/usr/bin/c++" "-DCMAKE_ASM_FLAGS= -ffunction-sections -fdat
a-sections -fPIC -m64" "-DCMAKE_ASM_COMPILER=/usr/bin/cc" "-DCMAKE_BUILD_TYPE=Debug"
  -- Configuring incomplete, errors occurred!

  --- stderr
  fatal: not a git repository (or any parent up to mount point /)
  Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
  CMake Error at CMakeLists.txt:3 (cmake_minimum_required):
    Compatibility with CMake < 3.5 has been removed from CMake.

    Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
    to tell CMake that the project requires at least <min> but has been updated
    to work with policies introduced by <max> or earlier.

    Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.



  thread 'main' panicked at /home/shum/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/cmake-0.1.54/src/lib.rs:1119
:5:

  command did not execute successfully, got: exit status: 1

  build script failed, must exit now
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: failed to compile `oculante v0.9.2`, intermediate artifacts can be found at `/home/shum/tmp/cargo-install4gkASx`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.

```

**Desktop (please complete the following information):**
 
Void Linux 

**Additional context**

Install solved by downgrading cmake to v3.30.x

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -102,7 +102,7 @@ webbrowser = "1.0"
 thiserror = "1.0.51"
 
 img-parts = "0.3" # for exif saving
-dark-light = "2.0"
+dark-light = { git = "https://github.com/rust-dark-light/dark-light", rev = "0f18d2fbcaa5d1c175db8aae7d53428988d7e961" } # 2.0 is broken from zbus change, temporarily using a git version until they release a new update.
 trash = "5.2"
 flate2 = "1.0" # for KTX
 bitflags = "2.8" # for KTX

